jerry-core/api/jerry.c

```diff
@@ -1538,6 +1538,20 @@ jerry_get_value_from_error (jerry_value_t value, /**< api value */
   return ret_val;
 } /* jerry_get_value_from_error */
 
+/**
+ * Set new decorator callback for Error objects. The decorator can
+ * create or update any properties of the newly created Error object.
+ */
+void
+jerry_set_error_object_update_callback (jerry_error_object_update_callback_t callback, /**< new callback */
```

Comment:
One thing I don't really like is the name. Why do we call this an "update" callback, when it's called upon object creation?

Reply:
The point is you can update (decorate) the created error object any way you want (e.g. add/remove/change properties). What name would you prefer?

Reply:
I think the name should reference the event that triggers the callback, instead of what you can do with it. (I mean, technically you could do anything that is permissibly by the API, not just update the object.)
How about calling it `error_object_created_callback` or something similar?

Reply:
Ok. Your suggestion is good. Patch is updated.